### PR TITLE
[NTOS:KE] Fix NtRaiseException and NtContinue on x64

### PR DESCRIPTION
**Pull request by tkreuzer**
## Purpose

Fix NtRaiseException and NtContinue on x64

## Tests
✅ KVM x86: https://reactos.org/testman/compare.php?ids=94981,94985,94988,94995,94999
- KVM x64: https://reactos.org/testman/compare.php?ids=94992,94996,94997,95000

### CODE REVIEW

File: ntoskrnl/ke/except.c [hunk 1]
```diff
@@ -223,29 +223,37 @@ NtContinue(IN PCONTEXT Context,
...
 
         /* Exit to new trap frame */
-        KiServiceExit2(TrapFrame);
+        KiExceptionExit(TrapFrame, ExceptionFrame);
     }
+#ifdef _M_IX86
     else
     {
         /* Exit with an error */
         KiServiceExit(TrapFrame, Status);
```
Comment (HBelusca) [inline ntoskrnl/ke/except.c:252]: Isn't it possible to have this thing in x64 as well, that does nothing? (This would avoid the 'ifdef' above the 'else')

**Comment by HBelusca** [comment]
In your 2nd commit message:
> - Implement NtRaiseException and NtContinue in asm, because they need an exception frame

Not sure I fully understand what you mean. It seems their code is still in C ? Except for maybe some stub entry and exit points.

**Comment by tkreuzer** [comment]
> In your 2nd commit message:
> 
> > * Implement NtRaiseException and NtContinue in asm, because they need an exception frame
> 
> Not sure I fully understand what you mean. It seems their code is still in C ? Except for maybe some stub entry and exit points.

Oh, the comment is outdated, I'll fix that.

File: ntoskrnl/ke/except.c (hunk 1 above)
Comment (tkreuzer) [reply ntoskrnl/ke/except.c:252]: That would just be confusing. We can probably remove the entire thing though.

**Review by HBelusca — APPROVED** [review]

**Review by DarkFire01 — APPROVED** [review]

**Pull request merged**